### ISSUE TITLE
Some actuion aren't showing cooldowns until refreshed

### ISSUE THREAD

**Issue by raysmith59** [brittyazel/Neuron#334]
For example, druid Renewal and Solar Beam.  When I use them it appears nothing happens, going into bar edit mode and back causes the cooldown to show up.

**Comment by brittyazel** [comment]
Strange... And it's just certain spells? I wonder why those spells aren't triggering the CD animation

**Comment by raysmith59** [comment]
To be clear, it's the numeric cooldown that isn't showing up that I'm expecting to.

Yeah, and in doing a bit more detailed testing it gets really weird.

For instance, when I hit my Moonkin button to shift from caster form to Moonkin form, these buttons do and don't show the GCD sweep (this isn't what I originally reported but I'm assuming these are all tied together somehow):

DO: Sunfire, Moonfire, Starsurge, Celestial Alignment, Innervate, Hibernate, Entangling Roots, Mass Entanglement, Soothe, Regrowth, the various form buttons
DON'T: Solar Wrath, Lunar Wrath, Starfall, Stellar Flare, Renewal, Solar Beam, Barkskin, Rebirth, Revive

Far as I'm concerned any button that's on the GCD should show the sweep and aside from maybe Renewal and Barkskin those are all on the GCD.

When I hit the same button again to shift back to caster form, NO buttons show the GCD sweep except the Moonkin button.  That doesn't make a lot of sense.

Those are all after first logging in.  If I go into and out of bar edit mode then most of the buttons show the GCD sweep as expected.  It's like something isn't fully loading or refreshing after login.

Here are the buttons that continue to not show the GCD sweep even after going into edit mode, these seem to confirm that buttons not on the GCD won't show the sweep normally:

Renewal, Solar Beam, Barkskin

Interestingly, in terms of my original report of buttons not showing the GCD, Barkskin DOES show it even when Renewal and Solar Beam don't.  Barkskin is also on a different action bar than the other two, based on that I tried moving buttons around - when they're on Action Bar 2 they show the GCD, when they're on Action Bar 1 they don't after logging in.  Best I can tell the bar configs are identical aside from location on the screen and Action Bar 1 being Shapeshift enabled (when I enable Shapeshift on bar 2 it still works correctly so unfortunately I don't think Shapesift is the issue) so no clue what's causing this.

**Comment by schmurfy** [comment]
I have the same issue but when it happens no cooldowns are showing, I tried disabling Neuron cooldown display and grabbed omnicc but interestingly omnicc has the same issue too with neuron's buttons.

**Comment by schmurfy** [comment]
I found more informations, as a druid here is how to trigger the issue:
- in bear form cast a spell with cooldown, it shows correctly
- cast "Travel Form"
- cast "Bear Form"
- the cooldown now longer show until I go into edit more (bar or button) and back

**Comment by brittyazel** [comment]
Yeah Blizzard changed something and didn't document it. I need to dig through their code and figure it out

**Comment by schmurfy** [comment]
I am not familiar with lua programming so I can't help much on it but I did some tests trying to help you narrow down this thing:

"Survival Instincts" and "Berserk" (4min and 3min cds) always manage to recover and the cooldown eventually shows itself again without actions from my part but all the smaller cooldowns I tried never show the cooldown again until I enter and leave edit mode.

This issue only happens on bar which are hidden at some points, it reproduced with alt state and multispec bars.

I hope you find what is happening, I really love this addon :)

**Comment by brittyazel** [comment]
Oh excellent info. That narrows it down a ton.

**Comment by brittyazel** [comment]
Can you test with 1.2.7 and let me know if you still have the issue?

**Comment by schmurfy** [comment]
it looks better, I did a few quick tests and the cooldown is not shown immediately if I switch to travel form, wait a few seconds, and switch back to bear form but it does come back quickly.

I will test more and report here.

**Comment by brittyazel** [comment]
Hrm. Strange. I triggered my survival instincts and switched between all my forms and it always showed the cooldown instantly

**Comment by schmurfy** [comment]
I have a few seconds before the cooldown appears not sure why but it is way more useable than before :)

**Comment by raysmith59** [comment]
Yeah, in very limited testing it isn't as bad as before but I was still seeing some issues that unfortunately appear to be random.  I'm going to put some play time in today to see if I can find some commonalities to report.
